proxy/src/main/java/com/velocitypowered/proxy/connection/registry/DimensionRegistry.java

```diff
@@ -0,0 +1,128 @@
+package com.velocitypowered.proxy.connection.registry;
+
+import java.util.HashSet;
+import java.util.NoSuchElementException;
+import java.util.Set;
+import javax.annotation.Nonnull;
+
+import net.kyori.nbt.CompoundTag;
+import net.kyori.nbt.ListTag;
+import net.kyori.nbt.Tag;
+import net.kyori.nbt.TagType;
+
+public class DimensionRegistry {
+
+  private final @Nonnull Set<DimensionData> dimensionRegistry;
```

Comment:
Do not use JSR-166 annotations (anything in `javax.annotation.*`). Since we follow Checker Framework, just drop the annotation (all fields are presumed to be non-null).

Reply:
As discussed on the discord, will do